inc/relationship/admin-filter.php

```diff
@@ -0,0 +1,279 @@
+<?php
+defined( 'ABSPATH' ) || die;
+
+use MetaBox\Support\Arr;
+
+class MBR_Admin_Filter {
+
+	const LIMIT              = 20;
+	const LIMIT_LABEL_OPTION = 50;
+
+	public function __construct() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		// Add filter for posts. Works only for posts. Terms & users don't have a similar filter.
+		add_action( 'restrict_manage_posts', [ $this, 'add_filter_for_posts' ] );
+
+		// Get options for autocomplete for select2 filter.
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_admin_script' ] );
+		add_action( 'wp_ajax_mbr_admin_filter', [ $this, 'ajax_get_options' ] );
+
+		// Process the filter: filter posts by relationships.
+		add_action( 'pre_get_posts', [ $this, 'filter_posts_by_relationships' ] );
+	}
+
+	public function add_filter_for_posts() {
+		$relationships = MB_Relationships_API::get_all_relationships();
+		array_walk( $relationships, [ $this, 'add_filter_select' ] );
+	}
+
+	private function add_filter_select( object $relationship ) {
+		$from = $relationship->from;
+		$to   = $relationship->to;
+
+		// Add filters for posts only.
+		if ( Arr::get( $from, 'object_type' ) !== 'post' && Arr::get( $to, 'object_type' ) !== 'post' ) {
+			return;
+		}
+
+		// Only show filters for current post type.
+		global $post_type;
+		if ( Arr::get( $from, 'field.post_type' ) !== $post_type && Arr::get( $to, 'field.post_type' ) !== $post_type ) {
+			return;
+		}
+
+		// Get data from or to relationship with current post type
+		$data = Arr::get( $from, 'field.post_type' ) === $post_type
+			? [
+				'data'     => $to,
+				'relation' => 'to',
+				'label'    => $from['meta_box']['title'],
+			]
+			: [
+				'data'     => $from,
+				'relation' => 'from',
+				'label'    => $to['meta_box']['title'],
+			];
+
+		if ( Arr::get( $data, 'data.admin_filter', false ) ) {
+			return;
+		}
```

Comment:
Check điều kiện này hình như bị sai em ạ. Cú pháp là như này `Arr::get( $array, $key, $default_value )`. Như em viết ở đây là nếu mà giá trị `admin_filter` lấy ra mà bằng `true` thì sẽ ko hiện filter => tức là sai. Đúng ra phải là ngược lại:

`if ( ! Arr::get( $data, 'data.admin_filter', false ) )`

Ngoài ra, ở đây em check `$data['data']` có vẻ như bị sai "phía" của relationship. VD: có 1 relationship từ Post tới Page. Mình đang muốn hiện filter ở màn hình của Post => Như thế cần phải lấy dữ liệu từ "phía" from. Nhưng `$data['data']` ở đây chứa dữ liệu từ "phía" `$to`.

Em check lại cẩn thận nhé.